core/src/main/java/com/linecorp/armeria/client/WebClientRequestPreparation.java

```diff
@@ -65,13 +67,33 @@ public final class WebClientRequestPreparation extends AbstractHttpRequestBuilde
     @Override
     public HttpResponse execute() {
         final HttpRequest httpRequest = buildRequest();
+        final RequestOptions requestOptions = buildRequestOptions();
+        return client.execute(httpRequest, requestOptions);
+    }
+
+    private RequestOptions buildRequestOptions() {
         final RequestOptions requestOptions;
+        final boolean requestStreaming = isRequestStreaming();
         if (requestOptionsBuilder != null) {
+            if (requestOptionsBuilder.exchangeType() == null) {
+                if (!requestStreaming) {
+                    requestOptionsBuilder.exchangeType(ExchangeType.RESPONSE_STREAMING);
+                }
+            }
             requestOptions = requestOptionsBuilder.build();
         } else {
-            requestOptions = RequestOptions.of();
+            if (!requestStreaming) {
+                requestOptions = RESPONSE_STREAMING_REQUEST_OPTIONS;
+            } else {
+                requestOptions = RequestOptions.of();
+            }
         }
-        return client.execute(httpRequest, requestOptions);
+        return requestOptions;
+    }
+
+    @SuppressWarnings("ReactiveStreamsUnusedPublisher")
```

Comment:
Commented on https://github.com/line/armeria/pull/4263#discussion_r908463283
But seems better to use `noinpsection` on the statement rather than the method annotation.